src/interface/desktop/main.js

```diff
@@ -151,44 +153,37 @@ function pushDataToKhoj (regenerate = false) {
 
     for (const syncedFile of lastSync) {
         if (!filesToPush.includes(syncedFile.path)) {
-            data.files.push({
-                path: syncedFile.path,
-                content: ""
-            });
+            fileObj = new Blob([""], { type: filenameToMimeType(syncedFile.path) });
```

Comment:
Files are currently being sent as `UploadFile` objects. Let's say we wanted to update this spec and allow the clients to send operation "types". For example, the operation actually being communicated here is a `delete`. It would be much cleaner, for intent, to relay that information by specifying it to be a `delete` op. Similarly, indexing a brand new file would be an `add`, an existing file an `update`. Would that be a heavy lift to modify the API specification here, or is that doable? Asking as the request format is already being updated.

Reply:
You can pass other fields in the request apart from the files to update. So while you can't pass the data in the UploadFile object directly you can pass in a dictionary that maps `filepath` to `operation-type`. Is that sufficient for these API spec updates?

Reply:
Yeah, that makes sense. We can add that as a follow-on change later.

Reply:
Sounds good